test/test_kernels.py

```diff
@@ -306,12 +318,25 @@ def test_p2e2p(actx_factory, base_knl, expn_class, order, with_source_derivative
         h_values = [1/2, 1/3, 1/5]
 
     rng = np.random.default_rng(19)
-    center = np.array([2, 1, 0][:knl.dim], np.float64)
-    sources = actx.from_numpy(
-        0.7 * (-0.5 + rng.random((knl.dim, nsources), dtype=np.float64))
-        + center[:, np.newaxis])
+
+    if isinstance(base_knl, HeatKernel):
+        # Setup sources so that there are no negative time intervals
+        center = np.array([0, 0, 0, 0.5][-knl.dim:], np.float64)
+        sources = (-0.5 + rng.random((knl.dim, nsources), dtype=np.float64)
+            + center[:, np.newaxis])
+        loc_center = np.array([0.0, 0.0, 0.0, 6.0][-knl.dim:]) + center
+        varying_axis = -1
+    else:
+        # The error behaviour is very specific to the point distribution.
+        # Even flipping the axis will lead to test failures
```

Comment:
Could you explain this comment/report more specifically what you saw? (And also confirm that you meant to add it where it is: i.e. not for the heat kernel?) Our existing multipoles/locals are reasonably well-tested, so I would be surprised if there was new, unexplained behavior there. But I'm curious what you saw.

Reply:
See the reproducer at https://github.com/inducer/sumpy/pull/187

Reply:
Removed this comment in 7d3b4440a78b53824c9578d21f59a1efbe73ab31. It was an issue with h_values as discussed.